pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -208,28 +212,42 @@ func statusForClaim(schedulingCtx *resourcev1alpha2.PodSchedulingContext, podCla
 // dynamicResources is a plugin that ensures that ResourceClaims are allocated.
 type dynamicResources struct {
 	enabled                    bool
+	fh                         framework.Handle
 	clientset                  kubernetes.Interface
 	claimLister                resourcev1alpha2listers.ResourceClaimLister
 	classLister                resourcev1alpha2listers.ResourceClassLister
 	podSchedulingContextLister resourcev1alpha2listers.PodSchedulingContextLister
+
+	// logger is only meant to be used by background activities which don't
+	// have some other logger in their parent callstack.
+	logger klog.Logger
 }
 
 // New initializes a new plugin and returns it.
 func New(plArgs runtime.Object, fh framework.Handle, fts feature.Features) (framework.Plugin, error) {
+	// TODO: the runtime should set up logging for each plugin, including
+	// adding a name for each one (same as in kube-controller-manager).
+	return NewWithLogger(klog.TODO(), plArgs, fh, fts)
```

Comment:
Do we have an issue for this TODO by the way?

Reply:
Not specifically for this one, but there are PRs pending for the conversion to contextual logging. Once those are complete, we can search for klog.TODO and fix it if not done yet.

Reply:
Seems better to pass-in a logger here, rather than a context like this https://github.com/kubernetes/kubernetes/pull/116884/files#diff-b5d92dd0897babb368df8c1aad9e95c957eca378bd46481d6d8501011f722715R77-R81

Reply:
Whatever gets done here must get coordinated when converting kube-scheduler to contextual logging. Let's not open that rathole right now.